bpm_projects_api/apis/dao.py

```diff
@@ -0,0 +1,85 @@
+class ProjectDAO(object):
+    def __init__(self):
+        self.counter = 0
+        self.projects = []
+
+    def get(self, id):
+        for project in self.projects:
+            if project['uid'] == id:
+                return project
+        return '', 404
+
+    def create(self, project):
+        self.counter += 1
+        project['uid'] = str(self.counter)
+        self.projects.append(project)
+        return project
+
+    def update(self, id, data):
+        project = self.get(id)
+        project.update(data)
+        return project
+
+    def delete(self, id):
+        project = self.get(id)
+        self.projects.remove(project)
+
+    def search(self, search_criteria):
+        matching_projects = self.select_matching_projects(search_criteria)
+
+        if len(matching_projects) > 0:
+            return matching_projects
+        else:
+            return '', 404
```

Comment:
The data layer should not be involved with the connection layer. This DAO is couple to rest results, even more with Flask RestPlus kind of results that returns a (result, http response code)